nucliadb/nucliadb/ingest/consumer/pull.py

```diff
@@ -131,64 +121,36 @@ async def loop(self):
-            )
-        else:
-            url = (
-                self.nuclia_cluster_url
-                + "/api/internal/processing/pull?partition="
-                + self.partition
-            )
-        async with aiohttp.ClientSession() as session:
+        if nuclia_settings.nuclia_service_account is not None:
+            headers["X-STF-NUAKEY"] = f"Bearer {nuclia_settings.nuclia_service_account}"
+
+        async with ProcessingHTTPClient() as processing_http_client:
             logger.info(f"Collecting from NucliaDB Cloud {self.partition} partition")
-            logger.info(f"{url}")
 
             while True:
                 try:
-                    async with session.get(
-                        url,
-                        headers=headers,
-                    ) as resp:
-                        if resp.status != 200:
-                            text = await resp.text()
-                            logger.exception(f"Wrong status {resp.status}:{text}")
-                            continue
-                        try:
-                            data = await resp.json()
-                        except Exception:
-                            text = await resp.text()
-                            logger.exception(f"Wrong parsing {resp.status}:{text}")
-                            continue
-
-                        if data.get("status") == "ok":
-                            check_proxy_telemetry_headers(resp)
-                            logger.info(
-                                f"Message received from proxy, partition: {self.partition}"
-                            )
-                            async with self.lock:
-                                try:
-                                    await self.handle_message(data["payload"])
-                                except Exception as e:
-                                    errors.capture_exception(e)
-                                    logger.exception(
-                                        "Error while pulling and forwarding proxy message to nucliadb nats"
-                                    )
-                                    raise e
-                        elif data.get("status") == "empty":
-                            logger_activity.debug(
-                                f"No messages waiting in partition #{self.partition}"
-                            )
-                            await asyncio.sleep(self.pull_time_empty_backoff)
-                        else:
-                            logger.info(f"Proxy pull answered with error: {data}")
-                            await asyncio.sleep(self.pull_time_error_backoff)
+                    data = await processing_http_client.pull(self.partition)
+                    if data.status == "ok":
+                        logger.info(
+                            f"Message received from proxy, partition: {self.partition}"
+                        )
+                        async with self.lock:
```

Comment:
I know it was already there but: why is the lock needed here?

Reply:
yeah, it's weird. I'm not sure either. Probably some legacy reason that no longer applies. I'm going to remove it.